web/src/lib/components/shared-components/scrollbar/scrollbar.svelte

```diff
@@ -23,25 +24,35 @@
   $: scrollY = toScrollY(timelineY);
 
   class Segment {
-    public count;
-    public height;
-    public timeGroup;
-
-    constructor({ count = 0, height = 0, timeGroup = '' }) {
-      this.count = count;
-      this.height = height;
-      this.timeGroup = timeGroup;
-    }
+    public count: number = 0;
+    public height: number = 0;
+    public timeGroup: string = '';
+    public date!: DateTime;
+    public hasLabel: boolean = false;
   }
 
-  $: segments = $assetStore.buckets.map(
-    (bucket) =>
-      new Segment({
-        count: bucket.assets.length,
-        height: toScrollY(bucket.bucketHeight),
-        timeGroup: bucket.bucketDate,
-      }),
-  );
+  const calculateSegments = (buckets: AssetBucket[]) => {
+    var height: number = 0;
+    var prev!: Segment;
+    return buckets.map((bucket) => {
+      var segment = new Segment();
+      segment.count = bucket.assets.length;
+      segment.height = toScrollY(bucket.bucketHeight);
+      segment.timeGroup = bucket.bucketDate;
+      segment.date = fromLocalDateTime(segment.timeGroup);
+
+      if (prev?.date.year != segment.date.year && (!prev || height > 16)) {
+        segment.hasLabel = true;
+        height = 0;
+      }
+
+      height! += segment.height;
```

Comment:
Since `height` has a default value (0), it should never be undefined, right? So '!' is unnecessary.